sogs/model/message.py

```diff
@@ -0,0 +1,65 @@
+from .. import db
+from ..db import query
+
+from .user import User
+
+
+class Message:
+    """Class representing a DM between users
+
+    Properties:
+        sender: sender user of the message
+        recip: recipant user of the message
+        data: opaque message data
+        signature: signature of data
+    """
+
+    def __init__(self, row=None, *, sender=None, recip=None, data=None, signature=None):
+        """
+        Constructs a Message from a pre-retrieved row *or* sender recipiant data and signature.
+        """
+        if row is None:
+            if None in (sender, recip, data, signature):
+                raise ValueError(
+                    "Message() error: provided no row but did not provide all members for construction"
+                )
+            for arg in (sender, recip):
+                if not isinstance(arg, User):
+                    raise ValueError("Message() error: sender or recipient was not a User model")
+
+            with db.transaction():
+                row = query(
+                    "INSERT INTO inbox(sender, recipient, data, signature) VALUES(:sender, :recipient, :data, :signature) RETURNING *",
```

Comment:
I think this is going to flake8 fail.  What I've been writing for these queries is:
```
            """
            INSERT INTO inbox (sender, recipient, data, signature)
            VALUES (:sender, ....)
            """,
```
Also needs `db.insert_and_get_pk` because `RETURNING *` isn't supported on non-super-recent sqlite.